lib/codegen/CreateTerm.cpp

```diff
@@ -21,6 +21,13 @@
 #include "llvm/Support/SourceMgr.h"
 #include "llvm/Support/raw_ostream.h"
 
+extern llvm::cl::OptionCategory CodegenCat;
+
+llvm::cl::opt<bool> ProofHintInstrumentation(
+    "proof-hint-instrumentation",
+    llvm::cl::desc("Enable instrumentation for generation of proof hints"),
+    llvm::cl::cat(CodegenCat));
```

Comment:
Yep, I basically agree with Guy's suggestion - let's push this option into the `ProofEvent` class. Otherwise the plumbing code etc. all looks good!